web/src/buddy_list.js

```diff
@@ -54,12 +60,26 @@ class BuddyListConf {
     }
 }
 
+function would_receive_message(user_id) {
+    const current_sub = narrow_state.stream_sub();
+    if (current_sub) {
+        return stream_data.is_user_subscribed(current_sub.stream_id, user_id);
+    }
+    const pm_ids_string = narrow_state.pm_ids_string();
+    if (pm_ids_string) {
+        const ids = pm_ids_string.split(",");
+        return ids.includes(user_id.toString());
+    }
+    return false;
+}
```

Comment:
It is, but slightly different because it's based on the narrow and not the composebox.

Reply:
Well you probably want a different name then -- something like `user_matches_narrow` or something -- since the name `would_receive_message` was chosen to refer to specifically if that user would receive the message currently being composed.

I also note that this has an `ids.includes` check in it -- some care may be needed here to use sets and thus avoid accidentally quadratic code.